lib/puppet/functions/fqdn_rand_string.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# @summary
+#   Generates a random alphanumeric string. Combining the `$fqdn` fact and an
+#   optional seed for repeatable randomness.
+#
+# Optionally, you can specify a character set for the function (defaults to alphanumeric).
+Puppet::Functions.create_function(:fqdn_rand_string) do
```

Comment:
ditto re namespacing

Reply:
Yeah, it is the plan :grin: I am currently playing three-cushion billiards:

1. Adjust non-deprecated functions so that they do not rely on deprecated functions; <--- we are here
2. Remove deprecated functions (since they have been deprecated for a long time);
3. "Break things" and namespace functions. <--- what you are talking about

The above is pretty clean in my head but I guess I should dump this somewhere where we can track progress, see what is coming next and so on.  I am really not good at this, and any suggestion is welcome!

Reply:
ack, feel free to close all comments related to this and we can deal with the general issue in #1346

Reply:
Great! I subscribed to the issue. Thanks!